File: 31.js
function evalRPN(tokens) {
  const stack = [];

  for (const token of tokens) {
    if (["+", "-", "*", "/"].includes(token)) {
      const b = stack.pop();
      const a = stack.pop();

      let res;
      if (token === "+") res = a + b;
      else if (token === "-") res = a - b;
      else if (token === "*") res = a * b;
      else res = a / b > 0 ? Math.floor(a / b) : Math.ceil(a / b); // truncate toward zero

      stack.push(res);
    } else {
      stack.push(parseInt(token));
    }
  }

  return stack[0];
}

// Example
console.log(evalRPN(["2", "1", "+", "3", "*"])); // 9
console.log(evalRPN(["4", "13", "5", "/", "+"])); // 6

File: 31.py
def evalRPN(tokens):
    stack = []
    for token in tokens:
        if token in "+-*/":
            b = stack.pop()
            a = stack.pop()
            if token == '+':
                stack.append(a + b)
            elif token == '-':
                stack.append(a - b)
            elif token == '*':
                stack.append(a * b)
            else:  # division (truncate toward zero)
                stack.append(int(a / b))
        else:
            stack.append(int(token))
    return stack[0]


# Example
print(evalRPN(["2", "1", "+", "3", "*"]))  # 9
print(evalRPN(["4", "13", "5", "/", "+"])) # 6
